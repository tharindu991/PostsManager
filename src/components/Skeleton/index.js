/*
 * Author: Tharindu Seneviratne
 * Date: 28/02/2020
 *
 * Skeleton component for render loading & refreshing states
 */

import React from 'react';
import PropTypes from 'prop-types';
import SkeletonPlaceholder from 'react-native-skeleton-placeholder';

const Skeleton = ({loading, refreshing, skeltonContent, children}) => {
  if (loading || refreshing) {
    return (
      <SkeletonPlaceholder speed={800}>{skeltonContent}</SkeletonPlaceholder>
    );
  } else {
    return children;
  }
};

Skeleton.propTypes = {
  loading: PropTypes.bool.isRequired,
};

export default Skeleton;
