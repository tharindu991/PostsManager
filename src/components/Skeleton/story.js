/*
 * Author: Tharindu Seneviratne
 * Date: 28/02/2020
 *
 *   Defines stories for skeleton component
 */

import React from 'react';
import {storiesOf} from '@storybook/react-native';
import Skeleton from './';
import {View} from 'react-native';

storiesOf('Skeleton', module).add('default view', () => (
  <Skeleton
    loading={true}
    skeltonContent={
      <View style={{flexDirection: 'row'}}>
        <View style={{width: 100, height: 100}} />

        <View
          style={{
            justifyContent: 'space-between',
            marginLeft: 12,
            flex: 1,
          }}>
          <View style={{width: '50%', height: 20}} />
          <View style={{width: '30%', height: 20}} />
          <View style={{width: '80%', height: 20}} />
        </View>
      </View>
    }
  />
));
