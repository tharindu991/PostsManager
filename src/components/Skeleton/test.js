/*
 * Author: Tharindu Seneviratne
 * Date: 28/02/2020
 *
 * Defines tests for Skeleton component
 */
import React from 'react';
import Skeleton from './';
import renderer from 'react-test-renderer';
import {View} from 'react-native';

jest.useFakeTimers();

describe('Skeleton Component', () => {
  it('Skeleton component renders correctly', () => {
    const component = renderer
      .create(
        <Skeleton
          loading={true}
          skeltonContent={
            <View style={{flexDirection: 'row'}}>
              <View style={{width: 100, height: 100}} />

              <View
                style={{
                  justifyContent: 'space-between',
                  marginLeft: 12,
                  flex: 1,
                }}>
                <View style={{width: '50%', height: 20}} />
                <View style={{width: '30%', height: 20}} />
                <View style={{width: '80%', height: 20}} />
              </View>
            </View>
          }
        />,
      )
      .toJSON();
    expect(component).toMatchSnapshot();
  });
});
