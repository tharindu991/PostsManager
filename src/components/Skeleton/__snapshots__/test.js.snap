// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Skeleton Component Skeleton component renders correctly 1`] = `
<View
  style={
    Object {
      "flexDirection": "row",
    }
  }
>
  <View
    style={
      Object {
        "position": "relative",
      }
    }
  >
    <View
      style={
        Array [
          Object {
            "height": 100,
            "width": 100,
          },
          Object {
            "backgroundColor": "#E1E9EE",
            "overflow": "hidden",
          },
        ]
      }
    >
      <View
        style={
          Object {
            "bottom": 0,
            "left": 0,
            "position": "absolute",
            "right": 0,
            "top": 0,
            "transform": Array [
              Object {
                "translateX": -350,
              },
            ],
          }
        }
      >
        <BVLinearGradient
          colors={
            Array [
              4292995566,
              4294113532,
              4292995566,
            ]
          }
          endPoint={
            Object {
              "x": 1,
              "y": 0,
            }
          }
          locations={null}
          startPoint={
            Object {
              "x": 0,
              "y": 0,
            }
          }
          style={
            Object {
              "flex": 1,
            }
          }
        />
      </View>
    </View>
  </View>
  <View
    style={
      Object {
        "flex": 1,
        "justifyContent": "space-between",
        "marginLeft": 12,
      }
    }
  >
    <View
      style={
        Object {
          "position": "relative",
        }
      }
    >
      <View
        style={
          Array [
            Object {
              "height": 20,
              "width": "50%",
            },
            Object {
              "backgroundColor": "#E1E9EE",
              "overflow": "hidden",
            },
          ]
        }
      >
        <View
          style={
            Object {
              "bottom": 0,
              "left": 0,
              "position": "absolute",
              "right": 0,
              "top": 0,
              "transform": Array [
                Object {
                  "translateX": -350,
                },
              ],
            }
          }
        >
          <BVLinearGradient
            colors={
              Array [
                4292995566,
                4294113532,
                4292995566,
              ]
            }
            endPoint={
              Object {
                "x": 1,
                "y": 0,
              }
            }
            locations={null}
            startPoint={
              Object {
                "x": 0,
                "y": 0,
              }
            }
            style={
              Object {
                "flex": 1,
              }
            }
          />
        </View>
      </View>
    </View>
    <View
      style={
        Object {
          "position": "relative",
        }
      }
    >
      <View
        style={
          Array [
            Object {
              "height": 20,
              "width": "30%",
            },
            Object {
              "backgroundColor": "#E1E9EE",
              "overflow": "hidden",
            },
          ]
        }
      >
        <View
          style={
            Object {
              "bottom": 0,
              "left": 0,
              "position": "absolute",
              "right": 0,
              "top": 0,
              "transform": Array [
                Object {
                  "translateX": -350,
                },
              ],
            }
          }
        >
          <BVLinearGradient
            colors={
              Array [
                4292995566,
                4294113532,
                4292995566,
              ]
            }
            endPoint={
              Object {
                "x": 1,
                "y": 0,
              }
            }
            locations={null}
            startPoint={
              Object {
                "x": 0,
                "y": 0,
              }
            }
            style={
              Object {
                "flex": 1,
              }
            }
          />
        </View>
      </View>
    </View>
    <View
      style={
        Object {
          "position": "relative",
        }
      }
    >
      <View
        style={
          Array [
            Object {
              "height": 20,
              "width": "80%",
            },
            Object {
              "backgroundColor": "#E1E9EE",
              "overflow": "hidden",
            },
          ]
        }
      >
        <View
          style={
            Object {
              "bottom": 0,
              "left": 0,
              "position": "absolute",
              "right": 0,
              "top": 0,
              "transform": Array [
                Object {
                  "translateX": -350,
                },
              ],
            }
          }
        >
          <BVLinearGradient
            colors={
              Array [
                4292995566,
                4294113532,
                4292995566,
              ]
            }
            endPoint={
              Object {
                "x": 1,
                "y": 0,
              }
            }
            locations={null}
            startPoint={
              Object {
                "x": 0,
                "y": 0,
              }
            }
            style={
              Object {
                "flex": 1,
              }
            }
          />
        </View>
      </View>
    </View>
  </View>
</View>
`;
